interactor/tests/complete_flow_tests.rs

```diff
@@ -0,0 +1,868 @@
+        amount: chain_interactor
+            .state
+            .get_initial_token_balance_for_address(
+                chain_interactor.user_address().clone().into(),
+                TokenIdentifier::from(&token.token_id),
+            )
+            - amount.clone(),
+    }];
+    chain_interactor
+        .check_address_balance(
+            &Bech32Address::from(chain_interactor.user_address.clone()),
+            expected_changed_user_balances,
+        )
+        .await;
+
+    let expected_mvx_balance = vec![TokenBalance {
+        token_id: token.token_id.clone(),
+        amount: amount.clone(),
+    }];
+    let mvx_esdt_safe_address = chain_interactor
+        .state
+        .get_mvx_esdt_safe_address(shard)
+        .clone();
+    chain_interactor
+        .check_address_balance(&mvx_esdt_safe_address, expected_mvx_balance)
+        .await;
+
+    chain_interactor
+        .execute_operation(
+            shard,
+            None,
+            Some(&token.token_id),
+            token.clone(),
+            amount.clone(),
+            None,
+        )
+        .await;
+
+    chain_interactor
+        .check_initial_wallet_balance_unchanged()
+        .await;
+
+    chain_interactor
+        .check_mvx_esdt_safe_balance_is_empty(shard)
+        .await;
+}
+
+/// ### TEST
+/// S-FORGE_COMPLETE-EXECUTE-SOVEREIGN-FLOW_OK
+///
+/// ### ACTION
+/// Deploy and complete setup phase, then call register token, execute operation and deposit sov token
+///
+/// ### EXPECTED
+/// The deposit is successful and the event is found in logs
+#[rstest]
+#[case::fungible(EsdtTokenType::Fungible, BigUint::from(ONE_HUNDRED_TOKENS))]
+#[case::non_fungible(EsdtTokenType::NonFungibleV2, BigUint::from(1u64))]
+#[case::semi_fungible(EsdtTokenType::SemiFungible, BigUint::from(ONE_HUNDRED_TOKENS))]
+#[case::meta_fungible(EsdtTokenType::MetaFungible, BigUint::from(ONE_HUNDRED_TOKENS))]
+#[case::dynamic_nft(EsdtTokenType::DynamicNFT, BigUint::from(1u64))]
+#[case::dynamic_sft(EsdtTokenType::DynamicSFT, BigUint::from(ONE_HUNDRED_TOKENS))]
+#[case(EsdtTokenType::DynamicMeta, BigUint::from(ONE_HUNDRED_TOKENS))]
+#[tokio::test]
+#[serial]
+#[cfg_attr(not(feature = "chain-simulator-tests"), ignore)]
+async fn test_register_execute_and_deposit_sov_token(
+    #[case] token_type: EsdtTokenType,
+    #[case] amount: BigUint<StaticApi>,
+) {
+    let mut chain_interactor = SovereignForgeInteract::new(Config::chain_simulator_config()).await;
+    let shard = SHARD_0;
+
+    chain_interactor
+        .deploy_and_complete_setup_phase(
+            DEPLOY_COST.into(),
+            OptionalValue::None,
+            OptionalValue::None,
+            None,
+        )
+        .await;
+
+    let token = chain_interactor.get_token_by_type(token_type);
+
+    let sov_token_id = TokenIdentifier::from_esdt_bytes(
+        REGISTER_TOKEN_PREFIX.to_string() + &token.token_id.clone(),
+    );
+    let token_display_name = TOKEN_DISPLAY_NAME;
+    let num_decimals = 18;
```

Comment:
18 decimals for NFT, for example, it doesn't look good. I don't know why it works. Maybe you can add a parameter in `#case` or use 0 for NFT/SFT/DyNFT/DySFT